### DISCUSSION TITLE
No official/documented certs REST API endpoint?

### DISCUSSION THREAD

**Discussion by danmichaelo** [vercel/community#16] [Help] [answered]
Is there no official REST API for listing and creating certs? 

https://api.vercel.com/v5/certs

seems to work, but https://vercel.com/docs/rest-api#endpoints/certs doesn't exist

**Comment by Pieparker** [comment, marked as answer]
Great observation, @danmichaelo! I believe that this API endpoint is undocumented as certificate generation should be automatic, as part of the project creation/custom domain addition flow.

Do you have a specific use case where certificate creation via the API is essential?

**Reply by danmichaelo** [reply]
Thanks for replying, @Pieparker ! Yes, I've described my use case at https://github.com/vercel/vercel/discussions/7273 since I first tried to get everything working using the CLI, but I'm now trying to use the REST API since it seems to offer more flexibility.

**Reply by Pieparker** [reply]
No problem! I'm glad that you were able to make use of the undocumented endpoint (essentially mimicking how our UI would make the calls). We tend not to document endpoints which don't fit a common use case though we often review this. We'll keep the CLI issue open and, fingers crossed, the team can take a look at what the issue may be there too!
